src/System.Net.Http/src/uap/System/Net/HttpHandlerToFilter.cs

```diff
@@ -47,9 +50,27 @@ internal HttpHandlerToFilter(RTHttpBaseProtocolFilter filter)
             }
             cancel.ThrowIfCancellationRequested();
 
-
             RTHttpRequestMessage rtRequest = await ConvertRequestAsync(request).ConfigureAwait(false);
-            RTHttpResponseMessage rtResponse = await _next.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+
+            RTHttpResponseMessage rtResponse;
+            try
+            {
+                rtResponse = await _next.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+            }
+            catch (TaskCanceledException)
+            {
+                throw;
+            }
+            catch (Exception)
+            {
+                Exception savedEx = (Exception)rtRequest.Properties[SavedExceptionLookupKey];
+                if (savedEx != null)
+                {
+                    throw savedEx;
```

Comment:
@stephentoub  So, I could change the `throw savedEx` here to `ExceptionDispatchInfo.Throw(savedEx)`.

But, since that exception is saved earlier in WinRTCertificateCallback, how can I save the exception w/ the right call stacks, etc. and then later throw it herer?  I don't understand ExceptionDispatchInfo object.